src/bun.js/test/jest.zig

```diff
@@ -1038,7 +1039,7 @@ pub const DescribeScope = struct {
             return err;
         }
 
-        if (comptime hook == .beforeAll or hook == .beforeEach) {
```

Comment:
Handling beforeAll here was causing all nested beforeAll blocks to run at the beginning of tests, rather than before their own describe blocks. Instead I created a separate function to run beforeAll